src/Datadog.Trace/Agent/Api.cs

```diff
@@ -70,6 +70,13 @@ public async Task SendTracesAsync(IList<List<Span>> traces)
                 }
                 catch (Exception ex)
                 {
+#if DEBUG
```

Comment:
Not really a fan of this but I don't know how else to surface the InvalidOperationException. Any suggestions?

Reply:
I think that having less forgiving failure conditions in development is a good thing for us. I'm not sure what the best pattern would be, but we could also choose to inspect logs and throw if there are error level logs.